### ISSUE TITLE
Air Nodes and the NPC's that refuse to use them; Thanks Gaben

### ISSUE THREAD

**Issue by Servius** [Facepunch/garrysmod-requests#1500]
Ive been doing some work with Player VS NPC maps and wanted to do something special with more "boss" npcs like Striders and Combine Choppers. I setup some strider nodes which are /supposed to/ tell striders where they can walk on the map. I also setup a net of standard air nodes across the map to allow for (or at least I thought it would allow for) full map navigation so hunting down players is easy and map wide. I even setup loops of air nodes in single chains to make it so they would regularly do loops around key buildings without causing that "AI Confusion" bug.  An issue arose when I finally got in game to test the new ainodes, the choppers(and all other air npcs like birds) won't use the air nodes and the strides won't use the strider nodes. They wont travel and hunt down enemies and I believe this behavior should be forced by default. It makes sense to make hl2 npcs feel like how they felt in the story without having to setup a func_tractrain in hammer for each individual interaction a player might have. 

TL;DR 
Force air npcs to use air nodes for navigation and to seek out and destroy enemy npcs that are visible. Force passive behavior so they feel more interactive.

**Comment by neico** [comment]
From VDC:
> You cannot force an NPC to use a hint (with the exception of actbusy routines). Hints are only suggestions, to be weighed by an NPC's AI code during gameplay.
> Sometimes AI will not use a hint when it might make sense to. This is a part of normal AI behavior.
Your "strider nodes" are actually hint nodes

There's `info_hint` which may cause AI confusion as it's not included in the nodegraph, it's merely for "visually interesting" situations.
There's also both `info_node_air` and `info_node_air_hint`, with the hint variant again being optional (tough it is still included in the nodegraph, basically a combination of `info_hint` and `info_node_air`)

You also need to keep in mind that a strider is considered a flying NPC for whatever reason, so you might need to use the air node variant instead... (don't quote me on that, it might be that it is both flying and non flying given the engine's quirks at times)

So this sounds like expected behavior until you state exactly which entities you are using.

**Comment by Servius** [comment]
The [Chopper](https://developer.valvesoftware.com/wiki/Npc_helicopter), [Crow](https://developer.valvesoftware.com/wiki/Npc_crow), [Seagull](https://developer.valvesoftware.com/wiki/Npc_seagull), etc. dont use the [air nodes](https://developer.valvesoftware.com/wiki/Info_node_air) for navigation as the Valve Wiki says they do. 

When comparing to a _working correctly_ npc like the [manhack, which is also listed by the Valve wiki as a "...Flying NPC, that uses info_node_air, info_node_air_hint or path_track's for AI navigation."](https://developer.valvesoftware.com/wiki/Npc_manhack) it is the only one that actually navigates and hunts down players as expected when spawned. The chopper, once spawned, will not exhibit any seek and destroy behaviors; just floats statically in place shooting at enemies that come within a radius of it, the birds do not fly around randomly throughout the world once spawned despite an air nodegraph being included in the .ain file, striders wont navigate throughout the map despite their nodes([yes their own info_node_air_hint with the "Strider Node" flag checked](https://developer.valvesoftware.com/wiki/Info_node_air_hint))being present. 

If you read the discussion section for [npc_helicopter](https://developer.valvesoftware.com/wiki/Npc_helicopter) youll see this: 
![image](https://user-images.githubusercontent.com/16316223/73327629-97f96980-420b-11ea-8d85-e370631f8141.png)

The issue is the lack of passive/expected behavior from a players perspective. Not from a Valve employees perspective. Because its more than likely they have the behavior available but left it disabled because they never ended up using it during their own work.

**Comment by Servius** [comment]
@kefta you mentioned elsewhere that the gmod team was focusing on replicating HL2 features and behaviors. 

Would increasing autonomy to actively display those behaviors fall under that new directive by the facepunch team?

**Comment by Kefta** [comment]
I don't work for Facepunch. The devs choose what issues are important.

**Comment by Servius** [comment]
Oh sorry! Ive been seeing you on these posts for like 2-5 years. Had no idea you werent employed by them, your responses and demeanor have always come off as experienced, mature, and informed all characteristics one might gleam from a conversation with someone like Rubat or Willox. 

Are you at least a moderator within the community? A volunteer representative?

**Comment by Kefta** [comment]
I only moderate the GMod Discord. It looks like what you're asking for is a way to call ``NPC::SetGoalEnt``/``NPC:NavSetGoalTarget`` (which sets the required goal ent, not a recommendation) and ``NPC:SetActivity(ACT_FLY)`` from maps, which is what ``npc_crow`` and ``npc_manhack`` use internally. The latter [isn't currently binded to Lua](https://github.com/Facepunch/garrysmod-requests/issues/1182), but you could use a ``lua_run`` ent to call ``NavSetGoalTarget`` on the desired NPC.

Adding this as default logic to Hammer would require a new node entity that is not optional to NPCs for flight paths, or some kind of intermediate logic entity that calls the aforementioned methods on an NPC automatically. Neither of these solutions will probably make their way into the default game so your best bet is a ``lua_run`` ent. You can use [Entity:SetSequence](https://wiki.facepunch.com/gmod/Entity:SetSequence) in-place of ``SetActivity`` for the meantime.
